### PR TITLE
gateio transfer can take symbols as fromAccount or toAccount, or take params.symbol

### PR DESCRIPTION
**Pull request by samgermain**
```
 % gateio transfer USDT 1 spot ADA/USDT
Debugger listening on ws://127.0.0.1:50019/958f3bb5-e279-46fd-9414-2fc95a1f444f
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
2022-04-28T10:02:01.684Z
Node.js: v14.17.0
CCXT v1.80.90
gateio.transfer (USDT, 1, spot, ADA/USDT)
2022-04-28T10:02:03.712Z iteration 0 passed in 241 ms

{
  id: undefined,
  timestamp: 1651140123710,
  datetime: '2022-04-28T10:02:03.710Z',
  currency: 'USDT',
  amount: 1,
  fromAccount: 'spot',
  toAccount: 'ADA/USDT',
  status: undefined,
  info: ''
}
```

```
% gateio transfer USDT 1 spot margin '{"symbol": "ADA/USDT"}'
2022-04-28T10:03:44.709Z
Node.js: v14.17.0
CCXT v1.80.90
gateio.transfer (USDT, 1, spot, margin, [object Object])
2022-04-28T10:03:46.727Z iteration 0 passed in 243 ms

{
  id: undefined,
  timestamp: 1651140226727,
  datetime: '2022-04-28T10:03:46.727Z',
  currency: 'USDT',
  amount: 1,
  fromAccount: 'spot',
  toAccount: 'margin',
  status: undefined,
  info: ''
}
```

```
% gateio transfer USDT 1 margin spot '{"symbol": "ADA/USDT"}'
2022-04-28T10:04:17.186Z
Node.js: v14.17.0
CCXT v1.80.90
gateio.transfer (USDT, 1, margin, spot, [object Object])
2022-04-28T10:04:19.223Z iteration 0 passed in 230 ms

{
  id: undefined,
  timestamp: 1651140259223,
  datetime: '2022-04-28T10:04:19.223Z',
  currency: 'USDT',
  amount: 1,
  fromAccount: 'margin',
  toAccount: 'spot',
  status: undefined,
  info: ''
}
```

```
% gateio transfer USDT 1 ADA/USDT spot                         
2022-04-28T10:04:44.896Z
Node.js: v14.17.0
CCXT v1.80.90
gateio.transfer (USDT, 1, ADA/USDT, spot)
2022-04-28T10:04:46.877Z iteration 0 passed in 235 ms

{
  id: undefined,
  timestamp: 1651140286877,
  datetime: '2022-04-28T10:04:46.877Z',
  currency: 'USDT',
  amount: 1,
  fromAccount: 'ADA/USDT',
  toAccount: 'spot',
  status: undefined,
  info: ''
}
```

```
 % gateio transfer USDT 1 spot swap    
2022-04-28T10:05:08.321Z
Node.js: v14.17.0
CCXT v1.80.90
gateio.transfer (USDT, 1, spot, swap)
2022-04-28T10:05:12.101Z iteration 0 passed in 226 ms

{
  id: undefined,
  timestamp: 1651140312101,
  datetime: '2022-04-28T10:05:12.101Z',
  currency: 'USDT',
  amount: 1,
  fromAccount: 'spot',
  toAccount: 'swap',
  status: undefined,
  info: ''
}
```

### CODE REVIEW

File: js/gateio.js [hunk 1]
```diff
@@ -3506,6 +3515,19 @@ module.exports = class gateio extends Exchange {
         });
     }
 
+    parseAccount (account) {
```
Comment (kroitor) [inline js/gateio.js:3518]: Can we avoid introducing `parseAccount` here?

Comment (samgermain) [reply js/gateio.js:3518]: The same logic is used for both toAccount and fromAccount, so we would have to write the code twice to avoid it

Comment (kroitor) [reply js/gateio.js:3518]: Let's clean out conflicts in this PR.

File: js/gateio.js [hunk 2]
```diff
@@ -3616,7 +3616,7 @@ module.exports = class gateio extends Exchange {
         if (fromId === 'margin' || toId === 'margin') {
             const symbol = this.safeString2 (params, 'symbol', 'currency_pair');
             if (symbol === undefined) {
-                throw new ArgumentsRequired (this.id + ' transfer requires a params.symbol argument for isolated margin transfers');
+                throw new ArgumentsRequired (this.id + " transfer requires params['symbol'] for isolated margin transfers");
```
Comment (samgermain) [inline js/gateio.js:3619]: Won't double quotes crash the build?

**Pull request merged**